internal/bundle/plain.go

```diff
@@ -0,0 +1,57 @@
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PlainV0 struct {
+	Objects []client.Object
+}
+
+func LoadPlainV0(bundle fs.FS) (PlainV0, error) {
+	const manifestsDir = "manifests"
+	var objects []client.Object
+	dirEntries, err := fs.ReadDir(bundle, manifestsDir)
+	if err != nil {
+		return PlainV0{}, err
+	}
+	if len(dirEntries) == 0 {
+		return PlainV0{}, errors.New("invalid bundle: found zero files")
+	}
+	for _, manifest := range dirEntries {
+		if manifest.IsDir() {
+			return PlainV0{}, fmt.Errorf("subdirectories are not allowed within the %q directory "+
+				"of the bundle image filesystem: found %q", manifestsDir, filepath.Join(manifestsDir, manifest.Name()))
+		}
+		manifestReader, err := bundle.Open(filepath.Join("manifests", manifest.Name()))
+		if err != nil {
+			return PlainV0{}, err
+		}
+		defer manifestReader.Close()
```

Comment:
Nit: `defer`-ing in a loop results in all of the deferred functions running at the end of the function rather than immediately after we've consumed that specific file. Can we extract this loop code into a separate function, basically:
```go
for _, manifest := range dirEntries {
	manifestObjects, err := getObjects(bundle, manifest)
	if err != nil {
		return nil, err
	}
	objects = append(objects, manifestObjects...)
}
```

That way, when you `defer` in `getObjects`, you end up actually closing the file before moving on to the next file.

Reply:
We could also potentially do 
```go
fs.WalkDir(bundle, manifestsDir, func(path string, d fs.DirEntry, err error) error {
	// handle individual dir entry in this closure.
})
```